### PR TITLE
Support for custom weekday names

### PR DESCRIPTION
**Pull request by JorgeAguirreLeon**
Added support for custom weekday names instead of default ones selected by moment (By using prop weekdayNames) 
You can see the result on the example

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,10 +23,8 @@
     "type": "git",
     "url": "https://github.com/onefinestay/react-daterange-picker"
   },
-  "peerDependencies": {
-    "react": ">=0.12.0"
-  },
   "dependencies": {
+    "react": ">=0.12.0",
```
Comment (jkimbo) [inline package.json]: Why was react moved from `peerDependencies`?

File: src/calendar/CalendarMonth.jsx [hunk 2]
```diff
@@ -76,10 +77,15 @@ const CalendarMonth = React.createClass({
 
   renderDayHeaders() {
     let {firstOfWeek} = this.props;
-    let indices = Immutable.Range(firstOfWeek, 7).concat(Immutable.Range(0, firstOfWeek));
+    let indices = Immutable.Range(firstOfWeek, 7).concat(Immutable.Range(0, Math.min(7, firstOfWeek)));
 
     let headers = indices.map(function(index) {
-      let weekday = WEEKDAYS.get(index);
+      let {weekdayNames} = this.props;
```
Comment (jkimbo) [inline src/calendar/CalendarMonth.jsx]: Can we not use `getDefaultProps` here?

File: src/calendar/CalendarMonth.jsx [hunk 3]
```diff
@@ -76,10 +77,15 @@ const CalendarMonth = React.createClass({
 
   renderDayHeaders() {
     let {firstOfWeek} = this.props;
-    let indices = Immutable.Range(firstOfWeek, 7).concat(Immutable.Range(0, firstOfWeek));
+    let indices = Immutable.Range(firstOfWeek, 7).concat(Immutable.Range(0, Math.min(7, firstOfWeek)));
```
Comment (jkimbo) [inline src/calendar/CalendarMonth.jsx]: Is `Math.min` necessary?

**Comment by jkimbo** [comment]
Like this! Sorry for taking so long to get back to you but if you could address the comments above I'll get it merged in.

File: src/calendar/CalendarMonth.jsx (hunk 3 above)
Comment (JorgeAguirreLeon) [inline src/calendar/CalendarMonth.jsx]: Hi, sorry it took me a few days to get back to you for such simple answers.
Absolutely right, this is not required.

File: package.json (hunk 1 above)
Comment (JorgeAguirreLeon) [inline package.json]: I wasn't sure then as I am not sure now, but `npm` gives a funny warning when using `peerDependencies` and this seems to solve the issue. [Warning info here] (https://docs.npmjs.com/files/package.json#peerdependencies)

File: src/calendar/CalendarMonth.jsx (hunk 2 above)
Comment (JorgeAguirreLeon) [inline src/calendar/CalendarMonth.jsx]: Yes! Yes we can!

**Comment by JorgeAguirreLeon** [comment]
Comments noted. Will fix 2/3 and wait for further feedback on the remaining one. I don't mind changing it but I'd like to understand the difference better.

File: package.json (hunk 1 above)
Comment (jkimbo) [inline package.json]: `peerDependencies` are a bit of a mess (hopefully npm3 will sort them out) but in the mean time if we could keep React as a peer it means people won't be loading two different versions of React with is a whole other world of pain.

**Comment by AlanFoster** [comment]
@JorgeAguirreLeon thanks for the PR; Are we able to progress with this work again? Let me know if you need a hand with anything :+1:

**Comment by JorgeAguirreLeon** [comment]
Hi, @AlanFoster I seem to have stumbled upon issues with React 0.14. 
I think I got it to work but need to remove the following line:

`React.initializeTouchEvents(true);`

I'm also having some issues with the whole
`npm run lint`
`npm run test`
They throw some nasty errors I haven't had time to fully investigate.

**Comment by AlanFoster** [comment]
@JorgeAguirreLeon thanks for the heads-up

If you rebase on master you'll see we've locked down our version of react to v12/v13. This should unblock you.

**Comment by nkato3** [comment]
Does anyone know the status of this? It would be really nice to have.

**Comment by AlanFoster** [comment]
For traceability; This work should also consider i18n support which should hopefully be released under #119 depending on the merge-order

@JorgeAguirreLeon Let me know if you have any problems rebasing this PR :+1:

**Pull request closed without merging**